### DISCUSSION TITLE
[Art competition] Canary

### DISCUSSION THREAD

**Discussion by DaehyeonAhn** [prosyslab-classroom/cs348-information-security#397] [Art competition]
![DALL·E 2023-05-22 07 12 13 - A woman who saw the ship leave the island sadly and hoped that the person in the ship would return safely, digital art](https://github.com/prosyslab-classroom/cs348-information-security/assets/126452243/f4a3a3be-f0ed-4f7c-9e69-f99451654b88)

## Short Description
For me, "Canary" means "A woman who saw the ship leave the island and hoped that the person who left would return safely".

## Long Description (up to 3 sentences)
We use canary to check for potential buffer overflows when receiving input or executing functions. I felt that this aspect is similar to a woman's heart as she sends off her beloved to the treacherous and unknown sea, hoping for his safe return. Just as we hopes for the canary to come back safely from the dangerous and uncertain functions. However, this story tragically ends because even if the canary returns safely, there is a suspicion that his heart may have changed.

**Comment by KihongHeo** [comment]
We need an automatic bug detector for her.
